src/Repository.ts

```diff
@@ -125,35 +174,60 @@ export class MongoRepository<T> {
     return document;
   }
 
+  /**
+   * Delete a record by ID
+   *
+   * @param {string} id
+   * @returns {Promise<any>}
+   * @memberof MongoRepository
+   */
   async deleteOneById(id: string): Promise<any> {
     return this.deleteOne({
       _id: new ObjectID(id)
     });
   }
 
+  /**
+   * Delete a record
+   *
+   * @param {*} conditions
+   * @returns {Promise<any>}
+   * @memberof MongoRepository
+   */
   async deleteOne(conditions: any): Promise<any> {
     const collection = await this.collection;
     return collection.deleteOne(conditions);
   }
 
+  /**
+   * Delete multiple records
+   *
+   * @param {*} conditions
+   * @returns {Promise<any>}
+   * @memberof MongoRepository
+   */
   async deleteMany(conditions: any): Promise<any> {
     const collection = await this.collection;
     return collection.deleteMany(conditions);
   }
 
   private createCollection(): void {
```

Comment:
Can you add a unit test for this method to confirm that calling it twice for the same collection does not fail?